Bring ckeditor back to base modules and sync modules

szeidler opened the pull request:
Integrate CKEditor library back to make file, to make default platform work and fix https://github.com/ramsalt/DrupalMakes/issues/12

CKEditor library will be dropped then out of the RamsaltBase make file.

Integrate RamsaltMedia additional CKEditor plugins. It will not hurt.

Sync some various patches and library versions with RamsaltBase repo, to be consistent.

szeidler committed:
Bring ckeditor back to base modules and sync modules

Integrate CKEditor library back to make file, to make default platform work and fix https://github.com/ramsalt/DrupalMakes/issues/12

Integrate RamsaltMedia additional CKEditor plugins. It will not hurt.

Sync some various patches and library versions with RamsaltBase repo, to be consistent.

esolitos commented on dists/base/base-d7-modules.make:
```diff
@@ -208,10 +211,40 @@ projects[zen][version] = "5"
 ; Amazon S3 cURL API
 libraries[s3-php5-curl][download][type] = "get"
-libraries[s3-php5-curl][download][url] = "https://github.com/tpyo/amazon-s3-php-class/archive/v0.5.1.zip"
+libraries[s3-php5-curl][download][url] = "https://github.com/tpyo/amazon-s3-php-class/tarball/master"
```
I had some issues at some point using the master tarball, why not use a release for libraries?

esolitos commented on dists/base/base-d7-modules.make:
```diff
@@ -227,13 +260,13 @@ libraries[backbone][type] = "library"
 ; Underscore.js (navbar)
 libraries[underscore][download][type] = "get"
-libraries[underscore][download][url] = "https://github.com/jashkenas/underscore/archive/1.5.2.zip"
+libraries[underscore][download][url] = "https://github.com/documentcloud/underscore/archive/master.zip"
```
Same as the s3-php library. :)

szeidler replied:
Good point. You're right. Master could lead to problems, when they introduce big changes at some point.

I just decided for one of the make files (here it was RamsaltBase)...

szeidler committed:
Use tagged versions of libraries

The master of 3rd-party libraries can get incompatible at a particular point of time. For that reason we should stick to tagged releases and update their definitions from time to time, after having checked the compatibility.

szeidler committed:
Fix accidental typo

esolitos commented:
+1

ooystein commented:
I have not had time to test this but it looks good. If marlon has approve it, let's merge it!

esolitos commented:
Thanks for the trust, beware that also my +1 is without testing as well, however as far as I can see it seems all good (compared to the related pull-req on bitbucket).
Plus, I am positive that Stephan did his tests. :)
